storm-core/src/jvm/backtype/storm/blobstore/FileBlobStoreImpl.java

```diff
@@ -0,0 +1,248 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package backtype.storm.blobstore;
+
+import backtype.storm.Config;
+import backtype.storm.utils.Utils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Very basic blob store impl with no ACL handling.
+ */
+public class FileBlobStoreImpl {
+  private static final long FULL_CLEANUP_FREQ = 60 * 60 * 1000l;
+  private static final int BUCKETS = 1024;
+  private static final Logger LOG = LoggerFactory.getLogger(FileBlobStoreImpl.class);
+  private static final Timer timer = new Timer("FileBlobStore cleanup thread", true);
+
+  public class KeyInHashDirIterator implements Iterator<String> {
+    private int currentBucket = 0;
+    private Iterator<String> it = null;
+    private String next = null;
+
+    public KeyInHashDirIterator() throws IOException {
+      primeNext();
+    }
+
+    private void primeNext() throws IOException {
+      while (it == null && currentBucket < BUCKETS) {
+        String name = String.valueOf(currentBucket);
+        File dir = new File(_fullPath, name);
+        try {
+          it = listKeys(dir);
+        } catch (FileNotFoundException e) {
+          it = null;
+        }
+        if (it == null || !it.hasNext()) {
+          it = null;
+          currentBucket++;
+        } else {
+          next = it.next();
+        }
+      }
+    }
+
+    @Override
+    public boolean hasNext() {
+      return next != null;
+    }
+
+    @Override
+    public String next() {
+      if (!hasNext()) {
+        throw new NoSuchElementException();
+      }
+      String current = next;
+      next = null;
+      if (it != null) {
+        if (!it.hasNext()) {
+          it = null;
+          currentBucket++;
+          try {
+            primeNext();
+          } catch (IOException e) {
+            throw new RuntimeException(e);
+          }
+        } else {
+          next = it.next();
+        }
+      }
+      return current;
+    }
+
+    @Override
+    public void remove() {
+      throw new UnsupportedOperationException("Delete Not Supported");
+    }
+  }
+
+  private File _fullPath;
+  private TimerTask cleanup = null;
+
+  public FileBlobStoreImpl(File path, Map<String, Object> conf) throws IOException {
+    LOG.info("Creating new blob store based in {}", path);
+    _fullPath = path;
+    _fullPath.mkdirs();
+    Object shouldCleanup = conf.get(Config.BLOBSTORE_CLEANUP_ENABLE);
```

Comment:
My mistake.  This is defined in Config.java.  Not sure why I had a compile error.